assets/src/edit-story/components/canvas/gridview/index.js

```diff
@@ -73,21 +91,56 @@ const Rectangle = styled.div`
   }
 `;
 
-function RangeControl({ value, onChange }) {
+// I think we can move this component to shared one in the future if we need this kind of spacer span.
+const SpacerSpan = styled.span`
+  margin-left: ${({ space }) => space || 8}px;
+`;
+
+function RangeControl({ value, onChange, min, max, step }) {
+  const updateRangeValue = (addition) => {
+    onChange(Math.min(max, Math.max(min, value + addition)));
+  };
+
+  let valueText;
+  switch (value) {
+    case max:
+      valueText = __('Large', 'web-stories');
+      break;
+    case min:
+      valueText = __('Small', 'web-stories');
+      break;
+    default:
+      valueText = __('Medium', 'web-stories');
+      break;
+  }
```

Comment:
This pretty much invalidates the generalization of this component though - if someone changes `min` or `max` without doing something about this, it kind of breaks the component.

I think `min` and `max` should just be hardcoded in the component now and if it needs to change, the component as a whole needs to change.

Also, the component should really be renamed to reflect, that it's not a generic range control, but a very specific thumbnail size range control (given the labels and icons, that are specific).

Reply:
Remove all props, define 3 variables on top and hard coded the values.

Reply:
@ndev1991 I think Morten here meant changing the component name to something that "reflects the specificity of this component", not changing it _literally_ to "Reflect".

I'd use something like 'ThumbnailSizeControl`

Reply:
Yes, `Reflect` is an even weirder name.

Reply:
🤦‍♂  how about ThumbnailSizeSlider or ThumbnailSizeControl? Something like GridViewSlider?